cmd/kube-controller-manager/app/controllermanager.go

```diff
@@ -369,11 +369,11 @@ func StartControllers(s *options.CMServer, kubeconfig *restclient.Config, rootCl
 	// Find the list of namespaced resources via discovery that the namespace controller must manage
 	namespaceKubeClient := client("namespace-controller")
 	namespaceClientPool := dynamic.NewClientPool(restclient.AddUserAgent(kubeconfig, "namespace-controller"), restMapper, dynamic.LegacyAPIPathResolverFunc)
-	groupVersionResources, err := namespaceKubeClient.Discovery().ServerPreferredNamespacedResources()
+	deletableGroupVersionResources, err := namespaceKubeClient.Discovery().ServerPreferredNamespacedResourcesWithVerbs("delete")
```

Comment:
namespace controller uses deletecollection, falling back to list+delete. I guess we need to call this twice and pass two sets of resources in?

```
deleteCollection = get("deletecollection")
listAndDelete = get("list","delete")
deleteCollection -= listAndDelete
listAndDelete -= deleteCollection
```
, once identifying the resources that support `deletecollection`, then again to find resources supporting `list`+`delete`, then remove